### PR TITLE
Fixing a bug in management client that is sending atom xml elements out of order

### PR DESCRIPTION
**Pull request by yvgopal**
Fixing a bug in management client that is sending atom xml elements in enity description
out of order in an update call. Because of this bug, deserializer on the service ignores
these out of order elements, and persists elements with default values in the righr order
resulting in these out of order elements getting duplicated in the resource xml.
This fix removes duplicate elements only if they are present from the xml, and sends
xml elements in the correct order in update call.

### CODE REVIEW

**Comment by JoshLove-msft** [comment]
Were any updates needed for rules?

File: sdk/servicebus/Microsoft.Azure.ServiceBus/src/Management/QueueDescriptionExtensions.cs [hunk 1]
```diff
@@ -26,14 +26,19 @@ public static XDocument Serialize(this QueueDescription description)
...
                 new XElement(XName.Get("EnableBatchedOperations", ManagementClientConstants.ServiceBusNamespace), XmlConvert.ToString(description.EnableBatchedOperations)),
                 description.AuthorizationRules?.Serialize(),
+                new XElement(XName.Get("IsAnonymousAccessible", ManagementClientConstants.ServiceBusNamespace), XmlConvert.ToString(description.IsAnonymousAccessible)),
                 new XElement(XName.Get("Status", ManagementClientConstants.ServiceBusNamespace), description.Status.ToString()),
                 description.ForwardTo != null ? new XElement(XName.Get("ForwardTo", ManagementClientConstants.ServiceBusNamespace), description.ForwardTo) : null,
                 description.UserMetadata != null ? new XElement(XName.Get("UserMetadata", ManagementClientConstants.ServiceBusNamespace), description.UserMetadata) : null,
+                description.internalSupportOrdering.HasValue ? new XElement(XName.Get("SupportOrdering", ManagementClientConstants.ServiceBusNamespace), XmlConvert.ToString(description.internalSupportOrdering.Value)) : null,
```
Comment (JoshLove-msft) [inline sdk/servicebus/Microsoft.Azure.ServiceBus/src/Management/QueueDescriptionExtensions.cs:33]: Do we not need the Xml label if the field is null?

File: sdk/servicebus/Microsoft.Azure.ServiceBus/src/Management/QueueDescriptionExtensions.cs [hunk 2]
```diff
@@ -26,14 +26,19 @@ public static XDocument Serialize(this QueueDescription description)
                 new XElement(XName.Get("MaxDeliveryCount", ManagementClientConstants.ServiceBusNamespace), XmlConvert.ToString(description.MaxDeliveryCount)),
                 new XElement(XName.Get("EnableBatchedOperations", ManagementClientConstants.ServiceBusNamespace), XmlConvert.ToString(description.EnableBatchedOperations)),
                 description.AuthorizationRules?.Serialize(),
```
Comment (JoshLove-msft) [inline sdk/servicebus/Microsoft.Azure.ServiceBus/src/Management/QueueDescriptionExtensions.cs:28]: In an example get response, I see that AuthorizationRules come after IsAnonymousAccessible.
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/tests/SessionRecords/ServiceBusManagementClientLiveTests/BasicQueueCrudOperationsAsync.json#L176

File: sdk/servicebus/Microsoft.Azure.ServiceBus/src/Management/QueueDescriptionExtensions.cs [hunk 3]
```diff
@@ -94,50 +101,56 @@ private static QueueDescription ParseFromEntryElement(XElement xEntry)
...
-                        qd.Status = (EntityStatus)Enum.Parse(typeof(EntityStatus), element.Value);
-                        break;
-                    case "AutoDeleteOnIdle":
-                        qd.AutoDeleteOnIdle = XmlConvert.ToTimeSpan(element.Value);
+                    case "AuthorizationRules":
+                        qd.AuthorizationRules = AuthorizationRules.ParseFromXElement(element);
                         break;
-                    case "EnablePartitioning":
-                        qd.EnablePartitioning = bool.Parse(element.Value);
+                    case "IsAnonymousAccessible":
```
Comment (JoshLove-msft) [inline sdk/servicebus/Microsoft.Azure.ServiceBus/src/Management/QueueDescriptionExtensions.cs:122]: Need to double check the order here between IsAnonymousAccessible and AuthRules. In the example linked above, they are swapped.

File: sdk/servicebus/Microsoft.Azure.ServiceBus/src/Management/QueueDescriptionExtensions.cs [hunk 4]
```diff
@@ -94,50 +101,56 @@ private static QueueDescription ParseFromEntryElement(XElement xEntry)
...
+                        break;
                     case "ForwardDeadLetteredMessagesTo":
                         if (!string.IsNullOrWhiteSpace(element.Value))
                         {
                             qd.ForwardDeadLetteredMessagesTo = element.Value;
                         }
                         break;
-                    case "AuthorizationRules":
-                        qd.AuthorizationRules = AuthorizationRules.ParseFromXElement(element);
+                    case "EnableExpress":
```
Comment (JoshLove-msft) [inline sdk/servicebus/Microsoft.Azure.ServiceBus/src/Management/QueueDescriptionExtensions.cs:152]: Do we need to parse EntityAvailabilityStatus?

**Comment by yvgopal** [comment]
No

---
In reply to: [720683278](https://github.com/Azure/azure-sdk-for-net/pull/16488#issuecomment-720683278) [](ancestors = 720683278)

File: sdk/servicebus/Microsoft.Azure.ServiceBus/src/Management/QueueDescriptionExtensions.cs (hunk 1 above)
Comment (yvgopal) [reply sdk/servicebus/Microsoft.Azure.ServiceBus/src/Management/QueueDescriptionExtensions.cs:33]: No

---
In reply to: [516215518](https://github.com/Azure/azure-sdk-for-net/pull/16488#discussion_r516215518) [](ancestors = 516215518)

File: sdk/servicebus/Microsoft.Azure.ServiceBus/src/Management/QueueDescriptionExtensions.cs (hunk 4 above)
Comment (yvgopal) [reply sdk/servicebus/Microsoft.Azure.ServiceBus/src/Management/QueueDescriptionExtensions.cs:152]: No. That is entity runtime information.

---
In reply to: [516222109](https://github.com/Azure/azure-sdk-for-net/pull/16488#discussion_r516222109) [](ancestors = 516222109)

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/servicebus/Microsoft.Azure.ServiceBus/src/Management/QueueDescription.cs [hunk 5]
```diff
@@ -273,11 +275,27 @@ public string UserMetadata
             }
         }
 
+        internal bool IsAnonymousAccessible { get; set; } = false;
+
+        internal bool SupportOrdering
+        { 
+            get
+            {
+                return this.internalSupportOrdering ?? !this.EnablePartitioning;
```
Comment (bainian12345) [inline sdk/servicebus/Microsoft.Azure.ServiceBus/src/Management/QueueDescription.cs:284]: >this.EnablePartitioning [](start = 56, length = 23)

I'm not sure what the relationship is between SupportOrdering and EnablePartitioning?

Comment (yvgopal) [reply sdk/servicebus/Microsoft.Azure.ServiceBus/src/Management/QueueDescription.cs:284]: If not partitioned, queues always support ordering. Topics optionally support ordering.
If partitioned, support ordering has no use. but it is part of the description.

File: sdk/servicebus/Microsoft.Azure.ServiceBus/src/Management/QueueDescription.cs [hunk 6]
```diff
@@ -308,6 +326,9 @@ public bool Equals(QueueDescription otherDescription)
                 && this.RequiresDuplicateDetection.Equals(other.RequiresDuplicateDetection)
                 && this.RequiresSession.Equals(other.RequiresSession)
                 && this.Status.Equals(other.Status)
+                && this.internalSupportOrdering.Equals(other.SupportOrdering)
```
Comment (JoshLove-msft) [inline sdk/servicebus/Microsoft.Azure.ServiceBus/src/Management/QueueDescription.cs:329]: Why would this not be:
```cs
this.SupportOrdering.Equals(other.SupportOrdering)
```
Won't this nullref if internalSupportOrdering is null?

**Pull request merged**